src/Microsoft.AspNetCore.Mvc.Razor/Internal/CompilerCacheResult.cs

```diff
@@ -87,5 +89,10 @@ public CompilerCacheResult(IList<IChangeToken> expirationTokens)
         /// Gets a delegate that creates an instance of the <see cref="IRazorPage"/>.
         /// </summary>
         public Func<IRazorPage> PageFactory { get; }
+
+        /// <summary>
+        /// Gets or sets a value that determines if a precompiled view was found.
+        /// </summary>
+        public bool IsPrecompiledView { get; set; }
```

Comment:
Since `CompilerCacheResult` is a struct, you'd have to accept this in a ctor. We don't use mutable structs in the most common use cases.